### ISSUE TITLE
Method for easily subsetting data driven tests would be useful

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#251]
Currently, we are fairly limited in the how we can run subsets of the data driven tests. It would be good to be able to specify the file(s) we want to apply the tests to, as the 1000 genomes files result in very long tests at the moment. Ideally, we'd like to add a simple command line option to nosetests, but it seems we'd need to write a standalone plugin for this. Other options would be to use a config file or environment variables, which would be easy enough to do.

Thoughts?

**Comment by diekhans** [comment]
See comment in other tickets.  IMHO, check in small, tailored SAM files.

**Comment by dcolligan** [comment]
Yeah, these tests (just ran them for the first time) take way too long.  What is the reason for having the data be so large (I'm assuming this is what is causing the test to run so long)?  Agree with @diekhans that smaller is better.

But if there's really no way getting around creating long-running tests, we can just tell nosetests to run only particular (fast) tests at launch.  Presumably the datadriven directory could be broken up into datadrivenUnit and datadrivenLong (or whatever structure we deem best) to accomodate this.

**Comment by diekhans** [comment]
Yes, you want to test as many paths as you can, not see how many times
a path can be execute.

Optional longer tests maybe useful.  But generally I fine that running
one-off larger tests to find new tests cases is better.

Large test for tracking performance changes maybe useful.

Mark

**Comment by jeromekelleher** [comment]
The problems are caused by 1000G VCFs, which are very tedious to work with. There's only 100 records in there, so we're not going overboard with lots of data. I think it's a good idea to have 1000G data in there, but if it turns out to be too slow then we can look at dumping some of the samples.

**Comment by dcolligan** [comment]
Has this been resolved in practice by modifying those VCF files?

**Comment by jeromekelleher** [comment]
The major pain has gone away by throwing away the majority of the 1000G samples, but it would still be nice to be able to specify the test and data file that we use. Suppose we've got 50 tests files, and a test is failing on one of them. You'll want to put in lots of debug prints and just run the test for the file in question.

This is a low priority though.

**Comment by dcolligan** [comment]
A hacky way to do this would just be to delete all the other data files, fix the problems in the test, and then use git to restore all of the files.

**Comment by jeromekelleher** [comment]
...  hadn't thought of that: nice hack!

We can close this issue if you like, I just thought it would be good to see if anyone had any bright ideas about how to do this cleanly.

**Comment by dcolligan** [comment]
I say leave it open; there should be a good way to do this cleanly.

**Comment by macieksmuga** [comment]
I like @dcolligan 's suggestion to lean on git to restore the tests to their original state after a debugging run. It works and satisfies Ockham's Razor - not introducing any further clever code to accomplish the goal.
